context-propagation/src/main/java/io/micrometer/context/ThreadLocalAccessor.java

```diff
@@ -38,13 +38,16 @@
     Object key();
 
     /**
-     * Return the current {@link ThreadLocal} value, or {@code null} if not set.
```

Comment:
I removed the mention of `null` signifying "not set". We actually can't tell whether the value was set or not. Implementations should perform a `ThreadLocal.get()`, which upon first retrieval, will initialise the storage and set it to the initialised value - either null, or provided via the `Supplier` from `ThreadLocal.withInitial(Supplier)`.